gobblin-hive-registration/src/main/java/gobblin/hive/HiveMetaStoreClientFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.hive;
+
+import org.apache.commons.pool.BasePoolableObjectFactory;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+
+/**
+ * An implementation of {@link BasePoolableObjectFactory} for {@link HiveMetaStoreClient}.
+ *
+ * @author ziliu
+ */
+public class HiveMetaStoreClientFactory extends BasePoolableObjectFactory<HiveMetaStoreClient> {
+
+  @Override
+  public HiveMetaStoreClient makeObject() throws Exception {
+    return new HiveMetaStoreClient(new HiveConf());
```

Comment:
Added a destroyObject method in `HiveMetaStoreClientFactory` which will take care of this.